lib/connectors/mongodb/connector.rb

```diff
@@ -0,0 +1,76 @@
+#
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License;
+# you may not use this file except in compliance with the Elastic License.
+#
+
+# frozen_string_literal: true
+
+require 'mongo'
```

Comment:
I totally missed this - I know that Tarek wrote a mongo connector for his spacetime, but when did adding a mongo connector become a priority for the product? I've never seen it in the list of potential connectors from PM (like with zoom, s3, assana, etc)

Reply:
Having a second connector built as we mature the BYOC protocol helps us a lot. It's a tech preview -- marked as such, it's not hurting anyone. In any case, I know that this backend is on the PM's radar for sure.

Reply:
MongoDB wasn't mentioned in the original project brief, but it's tied to some thinking from Matt as to the types of use cases we'd likely want to support going forward. 

We're actually seeing an interesting customer interest in MongoDB Atlas (https://www.mongodb.com/pricing) as well as some interesting use cases to see how we stack against Couchbase/Dynamo. MongoDB would not be ready in 8.4 to be considered ready as a native connector, but with our technical preview connector framework & service, it's a perfect proving ground to get earlier feedback. I foresee us doing a lot of technical preview connectors like this - early, out of release version releases into the repo, and then as usage and other metrics dictate, a promotion process to native. As part of this we'll want to harden up our definition of done to have a validation checklist to get connectors elevated to the native status, but in the mean time, it's great to sweeten the pot with this connector so PM can shop it around and use it to explore the new market use cases. 

FWIW, Tarek proposed it to the leads, and did get buyin from the PM team at the beginning of the sprint, and since connector service was progressing pretty well (thanks Chenhui!) we felt like we had some capacity to handle adding this as a stretch.